### PR TITLE
Add query pagination and limit for get questions

### PR DESCRIPTION
**Pull request by pchalubinska**
Added query string for pagination and created a limit on results. Created a new function to return all results.

### CODE REVIEW

File: server/controller/index.js [hunk 1]
```diff
@@ -18,8 +18,21 @@ exports.addQuestion = async (ctx) => {
   return ctx.body;
 };
 
+exports.getAllQuestions = async (ctx) => {
```
Comment (zsoltime) [inline server/controller/index.js]: We don't need two controllers for this, as we should never return every questions without a limit.

File: server/controller/index.js [hunk 2]
```diff
@@ -18,8 +18,21 @@ exports.addQuestion = async (ctx) => {
...
+  await Question.find({})
+    .then((res) => {
+      ctx.body = res;
+      return ctx.body;
+    });
+};
+
 exports.getQuestions = async (ctx) => {
+  const { page } = ctx.request.query;
+  const limit = 10;
```
Comment (zsoltime) [inline server/controller/index.js]: We should also accept a `limit` through `ctx.request.query` with a default value of 10.

File: server/controller/index.js [hunk 3]
```diff
@@ -18,8 +18,21 @@ exports.addQuestion = async (ctx) => {
...
+exports.getAllQuestions = async (ctx) => {
+  await Question.find({})
+    .then((res) => {
+      ctx.body = res;
+      return ctx.body;
+    });
+};
+
 exports.getQuestions = async (ctx) => {
+  const { page } = ctx.request.query;
```
Comment (zsoltime) [inline server/controller/index.js]: We should check if `page` is a number and either return an error message or revert to a default value of 1 maybe?

File: server/index.js [hunk 4]
```diff
@@ -29,7 +29,8 @@ router
   })
   .get('/api/questions', QuestionController.getQuestions)
   .get('/api/questions/random/:limit?', QuestionController.getRandomQuestions)
-  .get('/api/question/:id', QuestionController.getId);
+  .get('/api/question/:id', QuestionController.getId)
+  .get('/api/all-questions', QuestionController.getAllQuestions);
```
Comment (zsoltime) [inline server/index.js]: We don't need this one.

**Review by zsoltime — CHANGES_REQUESTED** [review]

**Review by zsoltime — APPROVED** [review]
Great job, @pchalubinska ;) Extra brownie points for checking the `res.length` 👍

**Comment by zsoltime** [comment]
@pchalubinska, I've just noticed that there is a merge conflict. It's really easy to solve. Do you want me to fix this, or will you give it a go?

**Comment by pchalubinska** [comment]
Yes I'll try:)

**Comment by pchalubinska** [comment]
I removed the hardcoded limit(10) which we don't need anymore. I think that solves the conflicts?

**Comment by zsoltime** [comment]
It does, thank you @pchalubinska :) It closes #56 and #57.

**Pull request merged**